### ISSUE TITLE
Searching by song artists 

### ISSUE THREAD

**Issue by kurros** [OxygenCobalt/Auxio#245]
#### Describe the bug:
Perhaps the work being done for issues #66 or #195 will resolve this, but if I attempt to search for a song on a compilation album, using the artist name, it will not appear in the search results. Searching for the song title works, with the artist name I tried to search for appearing in the result. 

#### Expected behavior
Searching for an artist to return results from compilation albums.

I did try with Ignore MediaStore Tags enabled as well. Auxio 2.6.3

**Comment by OxygenCobalt** [comment]
Auxio doesn't search the library by the artist name of a song. Currently, the app only searches by:

- Title
- Sort title
- File name

Could you elaborate further on what you want Auxio to search for given a query? I don't want to do a blind search for artists since it may clutter the results too much. However, if you limit the scope I may be more receptive to it.

**Comment by kurros** [comment]
I guess "Song file name" only applies when there is no metadata?

To be clear, a specific example: When search is set to All and I search for "tina turner", no results appear. I was expecting to find this result (abbreviated ffprobe):
```
Input #0, mp3, from 'Tina Turner_02_What's Love Got to Do with It.MP3':
  Metadata:
    title           : What's Love Got to Do with It
    artist          : Tina Turner
    album           : Sounds of the 80s
    album_artist    : Various Artists
    ARTISTS         : Tina Turner
```

I understand why Tina does not show up in the Artists tab (although that would be nice), but I expected to still be able to find her via search. Especially since the search result for "what's love" shows Tina Turner below the song title.

I appreciate the consideration of clutter and performance with find-as-you-type if the scope is expanded. That is an interesting thought about only expanding the search data for compilations. Maybe if Artist doesn't match Album Artist? That would also catch some soundtrack albums I have seen where Album Artist (and most of the tracks) are attributed to the composer, but there is a track or two from other artists.

**Comment by OxygenCobalt** [comment]
You say that:

> I understand why Tina does not show up in the Artists tab (although that would be nice)

This makes me think this is actually a XY problem and that you are really asking me to update Auxio's artist model to prefer the artist tag over the album artist tag. If I were to do that, you would have a "Tina Turner" entry that points to the song, and you would not have to search. 

If you still like the "Various Artists" entry, I could still keep that around through a more nuanced version of #195 that better differentiates artists and album artists. The details are fuzzy though.

Would you be fine with that @kurros? If yes, I'll be merging this into #195 and I'll think about how I want to implement your use case within the multi-artist system.

**Comment by OxygenCobalt** [comment]
I've implemented an MVP of this feature, and I actually really like it myself. I would prefer it much more than searching by song artist, so this is going to be closed and your issue (Tina Turner is not an artist) being merged into #195.
